doc: drivers: wifi: Document nRF71 Wi-Fi firmware interface (#29908)
nrfconnect/sdk-nrf: doc/nrf/drivers/wifi.rst

Wi-Fi drivers
#############

The |NCS| provides Wi-Fi® drivers that implement the IEEE 802.11 protocol and integrate the Nordic Semiconductor Wi-Fi devices with the Zephyr networking stack.

> These devices are FullMAC devices that implement the Wi-Fi protocol in the chipset.

Reviewer: Is this a well-defined terminology, understood by a Zephyr community developers?
If not, I would recommend simply explaining what components this FullMAC includes.

> The following sections describe the drivers available for each device series.

Reviewer: host drivers?

nRF70 Series
************

The nRF70 Series devices are Wi-Fi companion ICs that add Wi-Fi connectivity to Nordic Semiconductor System-on-Chips (SoCs), such as the nRF53, nRF54L, nRF54H, and nRF91 Series.
The following pages describe the nRF70 Series Wi-Fi driver, its OS-agnostic portable implementation, and the low-level API.

.. toctree::
   :maxdepth: 1

   wifi/nrf70_native
   wifi/nrf70_portable
   wifi/low_level_api

nRF71 Series
************

> The nRF71 Series is the next generation of Nordic Semiconductor Wi-Fi devices.

Reviewer: No need to refer it as a next generation. It is sufficient to describe this as what it is; a highly-integrated Wi-Fi SoC

Wi-Fi drivers
#############

The |NCS| provides host Wi-Fi® drivers that implement the IEEE 802.11 protocol and integrate the Nordic Semiconductor Wi-Fi devices with the Zephyr networking stack.

> These devices are FullMAC devices, which means that the complete IEEE 802.11 MAC layer, including the :abbr:`MLME (MAC Layer Management Entity)`, runs in the device firmware rather than on the host.

Reviewer 2: ```suggestion
These devices are FullMAC devices, which means that the complete IEEE 802.11 MAC layer, including the :abbr:`MLME (MAC Layer Management Entity)`, runs on the device firmware rather than on the host.
```

The host driver configures the device and transfers frames between the device and the networking stack.

> The following sections describe the host drivers available for each device series.

Reviewer 2: ```suggestion
The following sections describe the host drivers for each device series.
```

nRF70 Series
************

The nRF70 Series devices are Wi-Fi companion ICs that add Wi-Fi connectivity to Nordic Semiconductor System-on-Chips (SoCs), such as the nRF53, nRF54L, nRF54H, and nRF91 Series.
The following pages describe the nRF70 Series Wi-Fi driver, its OS-agnostic portable implementation, and the low-level API.

.. toctree::
   :maxdepth: 1

   wifi/nrf70_native
   wifi/nrf70_portable
   wifi/low_level_api

nRF71 Series
************

The nRF71 Series is a standalone, highly integrated, low-power Wi-Fi 6 and Bluetooth® Low Energy combo System-on-Chip (SoC).

> Unlike the nRF70 Series companion ICs, it integrates the Wi-Fi radio and the application processor in a single device, and therefore does not require a separate host SoC.

Reviewer 2: ```suggestion
The nRF71 Series integrates the Wi‑Fi radio and the application processor in a single device and therefore does not require a separate host SoC, unlike the nRF70 Series companion ICs.
```

Author: Combined this with @FrancescoSer's suggestion on the same sentence: I kept your explicit contrast with the nRF70 Series companion ICs, but split it into two sentences per the other suggestion. It now reads: "It also integrates the Wi-Fi radio and the application processor in a single device. Therefore, unlike the nRF70 Series companion ICs, it does not require a separate host SoC." Please confirm this works.

Author: Superseded by @ioannisg's suggestion on the same sentence, which keeps the nRF70 Series companion IC contrast. Resolving.

> Unlike the nRF70 Series companion ICs, it integrates the Wi-Fi radio and the application processor in a single device, and therefore does not require a separate host SoC.

Reviewer 3: ```suggestion
It also integrates the Wi-Fi radio and the application processor in a single device.
Therefore, it does not require a separate host SoC.
```

Author: Merged with @richabp's overlapping suggestion on the same sentence: used your two-sentence phrasing but retained the explicit contrast with the nRF70 Series companion ICs. It now reads: "It also integrates the Wi-Fi radio and the application processor in a single device. Therefore, unlike the nRF70 Series companion ICs, it does not require a separate host SoC." Please confirm this works.

Author: Superseded by @ioannisg's suggestion on the same sentence, which keeps the multi-sentence phrasing. Resolving.

> The following page describes the nRF71 Series driver and firmware interface.

Reviewer 2: ```suggestion
The following page describes the nRF71 Series driver and its firmware interface.
```

Wi-Fi drivers
#############

The |NCS| provides host Wi-Fi® drivers that implement the IEEE 802.11 protocol and integrate the Nordic Semiconductor Wi-Fi devices with the Zephyr networking stack.
These devices are FullMAC devices, which means that the complete IEEE 802.11 MAC layer, including the :abbr:`MLME (MAC Layer Management Entity)`, runs on the device firmware rather than on the host.
The host driver configures the device and transfers frames between the device and the networking stack.

> The following sections describe the host drivers for each device series.

Reviewer: ```suggestion
The following sections describe the host drivers for each Nordic device series.
```

nRF70 Series
************

The nRF70 Series devices are Wi-Fi companion ICs that add Wi-Fi connectivity to Nordic Semiconductor System-on-Chips (SoCs), such as the nRF53, nRF54L, nRF54H, and nRF91 Series.
The following pages describe the nRF70 Series Wi-Fi driver, its OS-agnostic portable implementation, and the low-level API.

.. toctree::
   :maxdepth: 1

   wifi/nrf70_native
   wifi/nrf70_portable
   wifi/low_level_api

nRF71 Series
************

The nRF71 Series is a standalone, highly integrated, low-power Wi-Fi 6 and Bluetooth® Low Energy combo System-on-Chip (SoC).

> It also integrates the Wi-Fi radio and the application processor in a single device.

Reviewer: ```suggestion
Unlike the nRF70 Series companion IC, nRF71 Series devices integrate the Wi-Fi radio and the application processor in a single SoC. The host Wi-Fi driver runs on the nRF71 Series application processor.
```

Author: Applied.

> Therefore, unlike the nRF70 Series companion ICs, it does not require a separate host SoC.

Reviewer: This sentence shall be removed, if the above suggestion is accepted

Author: Removed that sentence, since the L29 suggestion was applied.
